packages/notification-center/src/hooks/useUpdateUserGlobalPreferences.ts

```diff
@@ -0,0 +1,71 @@
+import type { IUserGlobalPreferenceSettings } from '@novu/client';
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+
+import { useFetchUserGlobalPreferencesQueryKey } from './useFetchUserGlobalPreferencesQueryKey';
+import { useNovuContext } from './useNovuContext';
+
+interface IUpdateUserGlobalPreferencesVariables {
+  preferences: { channelType: string; enabled: boolean }[];
+  enabled?: boolean;
+}
+
+export const useUpdateUserGlobalPreferences = ({
```

Comment:
I'm missing the hook and the controller code that allows to fetch global user preferences. I feel like we should return the global preferences together with template preferences, but for now, let's not show them in the NC. 
Then the `PreferenceLevelEnum.TEMPLATE` enum will be useful for filtering.